opus/application/static_media/js/opus.js

```diff
@@ -159,6 +189,89 @@ var opus = {
         o_search.allNormalizedApiCall().then(opus.getResultCount).then(opus.updatePageAfterResultCountAPI);
     },
 
+    // If the pasted url has selections, init opus.selections & opus.last_selections to the current hash's selections.
+    // This is like we init both opus.selections & opus.last_selections to {} when starting opus fresh.
+    InitSelections: function() {
+        let selections = o_hash.getSelectionsFromHash();
+        let modifiedSelections = {};
+        $.each(Object.keys(selections), function(idx, slug) {
+            // qtype is not updated in opus.selections
+            if (!slug.match(/qtype/)) {
+                modifiedSelections[slug] = selections[slug][0].replace("+", " ").split(",");
+            }
+        });
+        opus.last_selections = selections;
+        opus.selections = modifiedSelections;
+    },
+
+    // Check if two selections objects are the same, regardless of the order of array for each slug, e.g.:
+    // These two cases: obj1 & obj2 are the same:
+    // obj1 (convert from selections) = {planet: [mars, netpune], mission: [hubble]}
+    // obj2 (opus.selections) = {planet: [neptune, mars], mission: [hubble]}
+    // OR
+    // obj1 (convert from selections) = {planet: [mars, netpune]}
+    // obj2 (opus.selections) = {planet: [neptune, mars], mission: []}
+    // This is used to check if selections are changed manually by user or changed by actions done in opus
+    // Selections and opus.selections are the same when user makes selections in opus (url gets updated as well).
+    // If selections and opus.selections are not the same, it means the url is changed manually by user
+    checkIfSelectionsObjectsAreTheSame: function(obj1, obj2) {
```

Comment:
This entire function can't work at all.
- Two of the three return statements don't return a value, so the function will return "undefined" instead.
- The variable isTheSame isn't actually necessary, because you can always just "return True" or "return False".
- The loop doesn't catch the case of there being slugs in obj2 that aren't in obj1.
- I suggest you start with a simple slug list comparison...get the sorted list of keys from obj1 and obj2 and compare the two lists for equality. If they're not the same, then one of the two lists has a slug not in the other and you can return False.
- Otherwise, the slug list is guaranteed to be the same, so just iterate through and compared their values.